husky-communication/husky-communication-gen/src/main/java/org/projecthusky/communication/utils/HuskyUtils.java

```diff
@@ -0,0 +1,122 @@
+//	"ch-ppq1://%s?inInterceptors=%s&inFaultInterceptors=%s&outInterceptors=%s&outFaultInterceptors=%s&secure=%s&audit=%s&auditContext=%s&httpClientPolicy=%s"
+//	"ch-ppq2://%s?inInterceptors=%s&inFaultInterceptors=%s&outInterceptors=%s&outFaultInterceptors=%s&secure=%s&audit=%s&auditContext=#auditContext&httpClientPolicy=%s"
+//	"hpd-iti58://%s?inInterceptors=%s&inFaultInterceptors=%s&outInterceptors=%s&outFaultInterceptors=%s&secure=%s&audit=%s&auditContext=%s&httpClientPolicy=%s";
+//	"hpd-iti59://%s?inInterceptors=%s&inFaultInterceptors=%s&outInterceptors=%s&outFaultInterceptors=%s&secure=%s&audit=%s&auditContext=%s&httpClientPolicy=%s"
+	/**
+	 * Method to create the endpoint string for IPF.
+	 * 
+	 * @param transactionType the IHE transaction type
+	 * @param destination     the destination URI
+	 * @param auditEnabled    true if audit is enabled
+	 * @return the IPF/Camel endpoint string
+	 */
+	public static String createEndpoint(String transactionType, //
+			URI destination, //
+			boolean auditEnabled) //
+	{
+		return createEndpoint(transactionType, destination.toString(), auditEnabled);
+	}
+
+	/**
+	 * Method to create the endpoint string for IPF.
+	 * 
+	 * @param transactionType the IHE transaction type
+	 * @param destination     the destination as string
+	 * @param auditEnabled    true if audit is enabled
+	 * @return the IPF/Camel endpoint string
+	 */
+	public static String createEndpoint(String transactionType, //
+			String destination, //
+			boolean auditEnabled) //
+	{
+		boolean secure = destination.contains(HTTPS_LITERAL);
+		return createEndpoint(transactionType, destination.toString(), secure, auditEnabled);
+	}
+
+	/**
+	 * Method to create the endpoint string for IPF.
+	 * 
+	 * @param transactionType the IHE transaction type
+	 * @param destination     the destination as URI
+	 * @param secure          true if it should be secure
+	 * @param auditEnabled    true if audit is enabled
+	 * @return the IPF/Camel endpoint string
+	 */
+	public static String createEndpoint(String transactionType, //
+			URI destination, //
+			boolean secure, //
+			boolean auditEnabled) //
+	{
+		return createEndpoint(transactionType, destination.toString(), secure, auditEnabled);
+	}
+
+	/**
+	 * Method to create the endpoint string for IPF.
+	 * 
+	 * @param transactionType the IHE transaction type
+	 * @param destination     the destination as String
+	 * @param secure          true if it should be secure
+	 * @param auditEnabled    true if audit is enabled
+	 * @return the IPF/Camel endpoint string
+	 */
+	public static String createEndpoint(String transactionType, //
+			String destination, //
+			boolean secure, //
+			boolean auditEnabled) //
+	{
+		final String strippedUri = destination.replace(HTTPS_LITERAL, "").replace(HTTP_LITERAL, "");
```

Comment:
since java 8, the definition of final is not needed anymore, it can be removed due to effective final configuration.

Reply:
It does not harm to mark local variables as final.